gobblin-utility/src/test/java/gobblin/util/AvroFlattenerTest.java

```diff
@@ -0,0 +1,303 @@
+package gobblin.util;
+
+import org.apache.avro.Schema;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class AvroFlattenerTest {
```

Comment:
I think we should have some additional tests:
* Record within Record within Record
* Negative cases involving arrays or records

I think it will be easier to write tests if instead of inlining the schemas as string (lots of " escaping), schemas are loaded from resources: testX() reads the resources X.input.json and X.expected.json and verifies that flatten(X.input) == X.expected. Then we can easily add more tests.